src/Microsoft.DotNet.Internal.DependencyInjection.Testing/DependencyInjectionValidation.cs

```diff
@@ -37,10 +37,15 @@ public static class DependencyInjectionValidation
         Action<ServiceCollection> register,
         out string errorMessage,
         IEnumerable<Type> additionalScopedTypes = null,
-        IEnumerable<Type> additionalSingletonTypes = null)
+        IEnumerable<Type> additionalSingletonTypes = null,
+        IEnumerable<string> additionalExemptTypes = null)
```

Comment:
It is a bit wrong to require `IEnumerable` as we're enumerating over these 2+ times. We should either require `IReadOnlyCollection` or turn these into a list/array ourselves.

Reply:
changed everything to IReadOnlyCollection

Reply:
Be careful when making interface changes. This broke flow for Helix.

Reply:
sorry about this Stu, we should definitely be more mindful when making changes here